CRM/Extension/System.php

```diff
@@ -223,8 +224,8 @@ public function getManager() {
   public function getBrowser() {
     if ($this->browser === NULL) {
       $cacheDir = NULL;
-      if ($this->getDefaultContainer()) {
-        $cacheDir = $this->getDefaultContainer()->getBaseDir() . DIRECTORY_SEPARATOR . 'cache';
+      if (!empty($this->parameters['uploadDir'])) {
+        $cacheDir = $this->parameters['uploadDir'] . 'cache';
```

Comment:
Can we be sure we never need to add a directory separator here?

(@totten just drawing this patch to your attention in case you have any thoughts, looks good to me )